src/sentry/api/event_search.py

```diff
@@ -921,6 +921,8 @@ def get_json_meta_type(field_alias, snuba_type):
     if function_match:
         function_definition = FUNCTIONS.get(function_match.group(1))
         if function_definition and function_definition.get("result_type"):
+            if function_definition["result_type"] == "number" and snuba_type.startswith("DateTime"):
```

Comment:
If we make the max function's `result_type` None, I think the function will inherit the column's data type and you won't need this additional case.